numpy/core/include/numpy/ndarraytypes.h

```diff
@@ -1027,6 +1027,8 @@ typedef void (NpyIter_GetMultiIndexFunc)(NpyIter *iter,
  * eliminate overlap.
  */
 #define NPY_ITER_COPY_IF_OVERLAP            0x00002000
+/* At least one operand uses UPDATEIFCOPY semantics, use context manager */
+#define NPY_ITER_USE_CONTEXT_MANAGER        0x00004000
```

Comment:
Yeah, that is what I meant. I have not yet hashed it out in my brain what seems better. This approach does have some merit I guess (i.e. you can enter the same nditer twice). My first thought was to just wrap it all up and do `NpyIter_Deallocate` on `__exit__`, since I think `NpyIter_Deallocate` has to trigger the writing back for the normal C-Api usage.